### DISCUSSION TITLE
Waves Ver.15 Update

### DISCUSSION THREAD

**Discussion by mcaudio** [apohl79/audiogridder#1373] [Q&A]
Having a lot of trouble with AG and Waves ver. 15 update. Anyone else? Or workaround?

**Comment by mcaudio** [comment]
Nobody? Really?

**Comment by Kramer2025** [comment]
I am using an NLS chain and it is going bananas!  spikes only on VST client.
![ag waves](https://github.com/user-attachments/assets/0abdf079-3ad8-47d4-b55a-e9e20ac083cd)

**Reply by mcaudio** [reply]
Sorry you're having trouble. But you're ahead of me. I can't get Waves version 15 to work at all. Is that the only plugin failing for you? And is it version 15?

**Reply by Kramer2025** [reply]
It's the only plug-in I use in AG because usually they are very light on CPU. 
I keep the AG for Acustica or generally heavier on DSP plugins.
Yes it's V15. Have you tried to clear cache and fix them from waves center?

**Reply by mcaudio** [reply]
I keep all the Waves on server, and also on DAW machine for use when I "freeze" or render the file. I have cleared the cache, reinstalled both AG and waves central along with the plugins. Even tried offline installer. Nothing is working. Any luck on your end?

**Reply by Kramer2025** [reply]
I am on vacations now so there is no progress in that eradicated behavior on DSP meters. If I were you and have tried all that, I would just roll back to 14 to see how things are there. Assuming you don't need the V15. Can't think of anything else

**Comment by TimReago** [comment]
Looks like I am having the same problem with Waves v15.  I think it was working (or working better) under v14.  But I had not used it very much.  So not totally sure.  After installing v15 the other day.  Now most Waves plug-in's are not loading in to AG Server.  Has anyone compiled AG using VS 2022?  I have VS on the same computer.  Is there anything I need to do or know about stepping through the code to try and locate the problem?  For example, which cpp module should I focus on?  If so, please let me know, and I will try to report back.  Thanks, Tim

**Comment by mcaudio** [comment]
You're talking to the wrong guy with code! I stopped all that years ago, although my son works as a coder so I get good help when needed.

I'm still having the same problem. AG won't even recognize the Waves 15 plugins. Please let me know if you learn anything. And thanks!
________________________________

**Comment by TimReago** [comment]
Hi, 

I have not started to step through the code yet. I was hoping someone would be able to tell me where to set a break point, or at least which cpp file I should start focusing on to try to get Waves v15 working. I did drop back to Waves v14, and that is still working. I had copies of Waves v14 on offline install disks. 

I am working on a lot of computer code that is new to me right now, over 2 million lines of new code. My time is already way over scheduled. It really would helpful to know where to focus the attempt to discover why AG is not loading Wave v15. Poking around from scratch could cost a week or two or more of work, and I would prefer to use that time more wisely. If anyone can tell me which cpp file(s) I should focus on, or even better, which loader function name(s) to set break points on. Then I may be able to locate the problem within a reasonable amount of time. If so, I will let you know what I find. And I may or may not also some additional help in figuring out how to fix the problem. 

I have yet not looked at the AG code in any detail. But, just guessing, Waves may have changed it's file and/or interface format for v15. If so, some feature in VST3 may not have been included in AG that Waves started using as of v15. Like maybe a call to a VST3 interface function that is failing. 

It would be helpful to me, if I could get v15 up and running inside of AG. But if the fix is going to turn out to be too great of a drain on time, I may have to rely on v14. 

Thanks, 
Tim

**Comment by TimReago** [comment]
I have not started to step through the code yet.  I was hoping someone would be able to tell me where to set a break point, or at least which cpp file I should start focusing on to try to get Waves v15 working.  I did drop back to Waves v14, and that is still working.  I had copies of Waves v14 on offline install disks.

I am working on a lot of computer code right now, over 2 million lines of new code.  My time is already way over scheduled.  It really would be helpful to know where to focus the attempt to discover why AG is not loading Wave v15.  Poking around from scratch could cost a week or two or more of work, and I would prefer to use that time more wisely.  If anyone can tell me which cpp file(s) I should focus on, or even better, which loader function name(s) to set break points on.  Then I may be able to locate the problem within a reasonable amount of time.  If so, I will let you know what I find.  And I may or may not also need some additional help in figuring out how to fix the problem.

I have yet not looked at the AG code in any detail.  But, just guessing, Waves may have changed it's file and/or interface format for v15.  If so, some feature in VST3 may not have been included in AG that Waves started using as of v15.  Like maybe a call to a VST3 interface function that is now failing.

It would be helpful to me, if I could get v15 up and running inside of AG.  But if the fix is going to turn out to be too great of a drain on time, I may have to rely on v14.

Thanks,
Tim

**Reply by apohl79** [reply]
Plugin scanning happens mostly in JUCE. Take a look at Server::scanPlugin() in Server.cpp. AG server calls itself to scan a plugin. Check App.cpp for the cli interface. Not sure if a newer version of JUCE works better with the latest waves version, buts thats where you could get started. Upgrading JUCE requires porting some patches AG relies on, but it is usually not a big deal.

**Comment by TimReago** [comment]
I was able to install Waves V15 stand-alone plugins on Win 10 22H2 into the latest AudioGridderServer, including: regular Waves plugins and StudioVerse.  I tested on about 10 of the Waves plugins, and they appear to be accessible.  What I am thinking is that the Waves "Bundles" are where the V15 loader problem is.  May be something specific to the bundle packages, not yet being handled properly by JUCE.  Or maybe it might have been an installer error.  Not sure.  In any event, if you use V15 stand-alone plugins, it looks like they work in the current AudioGridder.

**Comment by mcaudio** [comment]
Incredibly cool of you to keep me in mind! Wow! You're a kind man!!!

Never occurred to me that it could be the bundle. I can see that being the issue. I'll try it.

I really appreciate this! I owe you one just for your kindness. Rare thing anymore. If I can do you any favors please reach out. I'll let you know how it works.

Michael
________________________________

**Comment by mcaudio** [comment]
Quick question. Have you tried re-installing the version 15 bundle? I went try installing the stand-alones and it occurred to me that I can install the version 14 bundle with no problem, yet fifteen hasn't worked. I'm going to try it tomorrow. Session tonight. I'll let you know what happens.
________________________________

**Comment by TimReago** [comment]
I don't think I tried to re-install V15 bundle.  I think I installed it once (or possibly twice) and it did not seem to work.

Although, as of tonight, I have seven of my cluster machines set up for running plugins.  And only one of the seven is doing something odd that is similar to the V15 bundle problem, where it is showing bundles highlighted in red and instead of individual plugins.  And for that one machine AG and the Waves plugins are not working as they are on the other six machines.

All of the machines have fresh installs of Win 10 22H2.  But some of the machines, including the one that is failing, were not a full reinstall where all Apps and Files were replaced.

That suggests to me, that if a bundle was previously installed, and even after a reinstall of Windows 10 + 22H2 update.  If Keep All Apps and Files was selected, and if an older version of the non-working V15 bundle was already present on that machine.  Then maybe it continues to be corrupted (non-loading).

**Comment by mcaudio** [comment]
Very ingenious and probably correct. It seems ridiculous that I can install UAD and Plugin Alliance bundles with no hitch and this is a persistent problem.

I do need a better server. Mine barely supports Win 10 but is still working very well. I don't really do much with virtual instruments (although moving more in that direction) so the machine is not heavily taxed. I think I will try a fresh Windows install and see what happens. Everything else has failed. I can't even get the version 15 stand-alone plugins to work.

Note: After uninstalling ver 14 I did notice Audiogridder was still looking for that version even after re-installing AG. Lends creedence to your theory!
________________________________

**Comment by TimReago** [comment]
I have some more info on this problem.  Even with a fresh install of Win 10, for larger Waves Bundles including those produced by installing multiple individual plugins into the same location, AG/JUCE still fails to load the plugins.  For my situation, 12 plugins was still too many even with a fresh install of Win 10.

Apparently the JUCE developers have been working on this problem:

https://forum.juce.com/t/juce-and-exception-handling/15144
https://forum.juce.com/t/any-way-to-speed-up-scanning-vst3-bundles/49038
https://forum.juce.com/t/vst3-shell-waves-plug-ins-instantiation-very-slow/33445

I will try to recompile AG using JUCE Version 8 to see if it solves the problem.  If there are not too many compile issues, hopefully that will work.

Another option:  Since apparently AG allows multiple plugin search paths.  I thought maybe installing only two or three Waves plugins to the default location.  Then moving the package to a sub-folder.  Then repeat for the next two or three Waves plugins.  Then listing each of the sub-folders in the AG settings dialog.  For a small number of plugins that are exclusively used via AG, this might work.

I'll try recompiling first.  If too many issues, I'll try splitting into multiple bundles.

**Comment by TimReago** [comment]
1) Unzipped audio_gridder into a top level folder, excluding JUCE folder.

2) Unzipped latest version of JUCE into the audio_gridder\JUCE folder.

3) Unzipped the Window dependency binaries into the audio_gridder\deps folder.

4) Run cmd.exe

5) audio_gridder> cmake.exe -G "Visual Studio 17 2022" -S . -B build

6) See ERRORS listing below.

7) Since I don't have enough experience with cmake and all of these dependencies to get this done quickly.  And since I am only interested in a small number of the Waves plugins, And since apparently the size of the Waves bundle is a factor.  I am going to try the second method of splitting up the bundles into 2 or 3 plugins each.  Then specifying multiple paths in AudioGridderServer settings dialog.

ERRORS:

-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
-- The C compiler identification is MSVC 19.38.33145.0
-- The CXX compiler identification is MSVC 19.38.33145.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: VS_2022/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: VS_2022/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could not find avformat
-- Could not find avdevice
-- Could not find avcodec
-- Could not find avutil
-- Could not find avfilter
-- Could not find swscale
-- Could not find swresample
-- Could not find FFMPEG
-- Could not find webp
-- Could not find webpmux
-- Could not find WebP
CMake Warning (dev) at CMakeLists.txt:85 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at CMake/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR) (Required is at least
  version "1.73.0")
Call Stack (most recent call first):
  CMake/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  CMake/share/cmake-3.30/Modules/FindBoost.cmake:2409 (find_package_handle_standard_args)
  CMakeLists.txt:85 (find_package)


-- Configuring incomplete, errors occurred!

**Comment by TimReago** [comment]
AudioGridder relies on JUCE.  JUCE relies on VST3 SDK.  But (apparently) JUCE does not fully implement the VST3 interface, including the bundled plugin's features.  And I am guessing that Waves bundles is using the VST3 bundles features.  So, it looks like I will be using VST3 SDK instead.  The VST3 SDK includes CMakeLists.txt files which can then be converted to .sln files for VS 2022.

**Comment by TimReago** [comment]
With a fresh install of Win 10 22H2, I tried using a sub-folder with a single plugin in the bundle for Waves v15 plugin.  AG server found the bundle in the sub-folder.  But AG server failed to locate the plugin inside of the bundle.  Whether it gets blamed on AG or JUCE, either way, it looks like the most sure solution is to drop back to VST3 SDK for bundled plugins.

**Comment by mcaudio** [comment]
Just Thinking about reaching out to see if you had learned anything new. Busy installing new motherboard and getting studio tweaked out. Really appreciate the word!

I'm going to address the issue again in the near future. May fail miserably but what the hell. Let you know what I learn, if anything. Talk to you soon and thank you!!

Michael
________________________________

**Comment by TimReago** [comment]
Still struggling with getting consistency out of Windows 10 22H2 installations and relative to AG's ability to access Wave's v15 plugins.  On some of the computers, they install and work.  And on others they completely do not work.  I feel like I need lower level access to the processes to be able to figure out what is going on.  Or at least to figure out what to do about it.  Working at the VST3 SDK level, I think will probably provide the most answers and ability to make the process consistent across a collection of machines.

I almost completely gave up on AG and JUCE tonight.  Until I realized a number of the computers are working.  But one of the main computers is not seeing the v15 plugins.  I did a full reinstall of the OS replacing everything, then the 22H2 upgrade.  Hopefully that will make the main computer consistent with the other computers.  I won't know until tomorrow where things stand.

But overall, my feeling is dropping down to the VST SDK level is probably the best chance at maintaining consistency.

**Comment by mihaimiculescu** [comment]
Is there a timer implemented somewhere in the detection process?
Something like "If it takes longer than <_whatever_> the detection process stops and reports a failure"?
Should such a logic exist in AG or Juce, simply assigning a longer timeout could do the trick?
_________________________________________
The reason I'm asking:
I repurposed my ancient HP Z400 machine (Win10) to work as AG server.
I still have the archaic Cakewalk Sonar on it. The Sonar VST scanner took a very long time, smth. like 20 min., to fully detect the v15 bundles, but it finally did it.
AG on the other hand gave up rather quickly. It successfully detected only the smallest existing bundle and blacklisted the rest.
________________________________________
@apohl79  Any ideas?

**Comment by TimReago** [comment]
Because other DAW's are working, I suspect either AG or JUCE has a problem.  I did find a few JUCE discussion threads that seemed to be talking about the problem.  It looks like I referenced some of them above.

Maybe a newer version of JUCE would solve the problem.  But I am not up to speed enough to be able to quickly swap the current AG version of JUCE for the most recent.

Some of my machines are working with AG and I think also with v15 individual plugins.  But other virtually identical machines are not working.

It seems really hard to diagnose without being more familiar with the source code and how to successfully compile and debug it.

**Comment by mcaudio** [comment]
Talk about not up to speed. My skills have been waning since Windows 3! Made life too easy.

I've pretty much come to the same conclusion. Even tried it on Mac. Gotta be third party.

I'm going to talk to my son about JUCE. He's a coder and maybe can come up with a fix. Stay in touch and I'll let you know what we find out.

Really appreciate you keeping me up to date. It's going to be a while but maybe he can come up with something.

Michael
________________________________
